sdk/core/ci.yml

```diff
@@ -42,6 +42,56 @@ stages:
       LiveTestTimeoutInMinutes: 90 # default is 60 min. We need a little longer on worst case for Win+jsonTests
       LineCoverageTarget: 93
       BranchCoverageTarget: 55
+      PreTestSteps:
+        - task: UsePythonVersion@0
+          displayName: 'Use Python 3'
+          inputs:
+            versionSpec: '3'
+          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+
+        - pwsh: |
+            python --version
+            pip install -r requirements.txt
+          workingDirectory: build/sdk/core/azure-core/test/ut
+          displayName: Install Python requirements.
+          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+        - task: PowerShell@2
+          displayName: 'Launch python websocket server'
+          inputs:
+            pwsh: true
+            filePath: build/sdk/core/azure-core/test/ut/Start-WebSocketServer.ps1
+            arguments: $(Build.SourcesDirectory)/WebSocketServer.log
+            workingDirectory: build/sdk/core/azure-core/test/ut
+          condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+          # It would be nice to collapse this branch with the previous one, but nohup doesn't seem to
+          # behave when called from powershell.
+        - bash: |
+            nohup python sdk/core/azure-core/test/ut/websocket_server.py > $(Build.SourcesDirectory)/WebSocketServer.log &
+          workingDirectory: build
+          condition: and(succeeded(), ne(variables['Agent.OS'], 'Windows_NT'), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+          displayName: Launch python websocket server (Linux).
+      PostTestSteps:
```

Comment:
Will this post test step run even if the pipeline fails and terminates in the middle, before hitting this step, for whatever reason (say network error, some unit test failure, etc.)?

Reply:
Yes.